### PR TITLE
collections: update docs of slice get() and friends

### PR DESCRIPTION
**Pull request by birkenfeld**
Resubmit of #38216.

r? @GuillaumeGomez 

BTW, instead of closing a PR just because it is old and the team member who offered to fix it up did not have the time to do so, why not ping them instead? (cc @alexcrichton)

### CODE REVIEW

**Comment by GuillaumeGomez** [comment]
Just reopen the PR and talk about it with the one who closed it. Generally we ask a first time before closing, but it depends on the member.

File: src/libcollections/slice.rs [hunk 1]
```diff
@@ -342,15 +342,22 @@ impl<T> [T] {
         core_slice::SliceExt::last_mut(self)
     }
 
-    /// Returns the element of a slice at the given index, or `None` if the
-    /// index is out of bounds.
+    /// Returns a reference to an element or subslice depending on the type of
+    /// index.
+    ///
+    /// - If given a position, returns a reference to the element at that
+    ///   position or `None` if out of bounds.
```
Comment (GuillaumeGomez) [inline src/libcollections/slice.rs]: Can you add `None` url please?

File: src/libcollections/slice.rs [hunk 2]
```diff
@@ -342,15 +342,22 @@ impl<T> [T] {
...
+    /// - If given a range, returns the subslice corresponding to that range,
+    ///   or `None` if out of bounds.
```
Comment (GuillaumeGomez) [inline src/libcollections/slice.rs]: Can you add `None` url please?

File: src/libcollections/slice.rs [hunk 3]
```diff
@@ -360,7 +367,10 @@ impl<T> [T] {
         core_slice::SliceExt::get(self, index)
     }
 
-    /// Returns a mutable reference to the element at the given index.
+    /// Returns a mutable reference to an element or subslice depending on the
+    /// type of index (see [`get()`]) or `None` if the index is out of bounds.
```
Comment (GuillaumeGomez) [inline src/libcollections/slice.rs]: Can you add `None` url please?

**Review by GuillaumeGomez — CHANGES_REQUESTED** [review]
Just a few urls missing.

**Comment by birkenfeld** [comment]
Did you look at the closed PR? There is a reason why I did not put in the URLs.

**Comment by birkenfeld** [comment]
> Just reopen the PR and talk about it with the one who closed it.

That's what I did, no?

**Comment by GuillaumeGomez** [comment]
No I mean, the PR that been closed. You created a new one. However, maybe it wasn't possible to just reopen it? Not sure...

**Comment by GuillaumeGomez** [comment]
Oh damn you're right. And no, I didn't look at the original PR before you said it. Then it's all good. Sorry for the incovenience.

@bors: r+ rollup

**Comment by birkenfeld** [comment]
Thanks! Sorry if I sounded a bit frustrated ;)

**Comment by GuillaumeGomez** [comment]
It's understandable. Just try to speak with people if you have the feeling things aren't moving. There are a lot of PRs and work done but not always publicly.

**Pull request merged**